maths/trapezoidal_rule.py

```diff
@@ -20,29 +20,70 @@ def method_1(boundary, steps):
     y += (h / 2.0) * f(a)
     for i in x_i:
         # print(i)
-        y += h * f(i)
+        y += h * f(func, i)
     y += (h / 2.0) * f(b)
     return y
 
 
 def make_points(a, b, h):
-    x = a + h
-    while x < (b - h):
+    """
+    Yields equally spaced points between a and b
+    >>> x_i = make_points(0, 0.4, 0.1)
+    >>> next(x_i)
+    0.1
+    >>> next(x_i)
+    0.2
+    >>> next(x_i)
+    0.3
+    >>> next(x_i)
+    Traceback (most recent call last):
+        ...
+    StopIteration
+
+    >>> x_i = make_points(0, 1, -0.2)
+    >>> next(x_i)
+    Traceback (most recent call last):
+        ...
+    ValueError: h must be positive
+
+    >>> x_i = make_points(1,0,0.2)
+    >>> next(x_i)
+    Traceback (most recent call last):
+        ...
+    ValueError: a must be less than b
+    """
+    if h <= 0:
+        raise ValueError("h must be positive")
+    if a >= b:
+        raise ValueError("a must be less than b")
+
+    x = round(a + h, 10)
+    while x <= (b - h):
         yield x
-        x = x + h
+        x = round(x + h, 10)
 
 
-def f(x):  # enter your function here
-    y = (x - 0) * (x - 0)
-    return y
+def f(y, x):  # enter your function here
+    """
+    Returns the value of a lambda functiοn y at x
+    >>> f(lambda x: x**2,2)
+    4
+    >>> f(lambda x: x**2,-1)
+    1
+    >>> f(lambda x: (x+1)/2,5)
+    3.0
+    >>> f(lambda x: (x+1)/2,0)
+    0.5
+    """
+    return y(x)
```

Comment:
See previous comment: why have a wrapper function for a function rather than just running the function directly?